support-frontend/assets/pages/digital-subscription-checkout/helpers/payPal.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+import { Action, setPayPalHasLoaded } from 'pages/digital-subscription-checkout/digitalSubscriptionCheckoutActions';
+import { loadPayPalRecurring, payPalRequestData } from 'helpers/paymentIntegrations/payPalRecurringCheckout';
+import type { ContributionType } from 'helpers/contributions';
+import { State } from 'pages/digital-subscription-checkout/digitalSubscriptionCheckoutReducer';
+import { billingPeriodFromContrib } from 'helpers/contributions';
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import type { Csrf } from 'helpers/csrf/csrfReducer';
+import { logException } from 'helpers/logger';
+import * as storage from 'helpers/storage';
+import { finalPrice as dpFinalPrice } from 'helpers/productPrice/digitalProductPrices';
+
+
+const showPayPal = (dispatch: Function) => {
+  loadPayPalRecurring()
+    .then(() => {
+      dispatch(setPayPalHasLoaded());
+    });
+};
+
+const setupPayPalPayment = (
+  resolve: string => void,
+  reject: Error => void,
+  currency: IsoCurrency,
+  csrf: Csrf,
+  contributionType: ContributionType,
+) =>
+  (dispatch: Function, getState: () => State): void => {
+    const state = getState();
+    const csrfToken = csrf.token;
+    const { price } = dpFinalPrice(
+      state.page.checkout.productPrices,
+      state.page.checkout.billingPeriod,
+      state.common.internationalisation.countryId,
+    );
+    const billingPeriod = billingPeriodFromContrib(contributionType);
+    storage.setSession('selectedPaymentMethod', 'PayPal');
+    const requestBody = {
+      amount: price,
+      billingPeriod,
+      currency,
+    };
+
+    fetch('/paypal/setup-payment', payPalRequestData(requestBody, csrfToken || ''))
```

Comment:
Maybe this bit could go in a shared library rather than per-page action? Since it's the same as in `contributionsLandingActions.js`

Reply:
Yeah good point, I'll refactor it now